### ISSUE TITLE
Refined asynchronous initialization

### ISSUE THREAD

**Issue by jklmnn** [Componolit/gneiss#140]
Currently the initialization callback is merged with the event callback. The original thought was to handle all initializations in the same callback. This turned out to be impractical. The new approach is to have a specific initialization callback for each session:

```
generic
   with procedure Initialized (Session : in out Client_Session);
procedure Initialize (Session : in out Client_Session; Cap : Capability; ...);
```

The `Session_State` will be removed in favor of a `Initialized` and a `Pending` function. The latter is `Ghost`. When `Initialize` is called the session is implicitly pending. In this state it cannot be interacted with. When the callback has been called it will not be pending anymore.

**Comment by jklmnn** [comment]
The ghost pending function prevents early failure as it would require a callback after returning from the current procedure even when the initialization fails in the first check. The `Pending` state will however not require `Initialize` to be called again. Also a specific callback just for initialization is preferable.

**Comment by jklmnn** [comment]
Fixed by #141.
